cmd/template-script-backup-restore/main.go

```diff
@@ -0,0 +1,206 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2024 Red Hat, Inc.
+ *
+ */
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"strings"
+
+	"kubevirt.io/client-go/log"
+)
+
+const (
+	// TODO: replace value with your solution namespace
+	defaultNamespace = ""
+)
+
+var (
+	backupNS         string
+	includedNS       string
+	includeResources string
+	selector         string
+	snapshotLocation string
+	verify           bool
+	deleteBackup     bool
+	backupName       string
+	restoreName      string
+)
+
+//TODO: update flags default if needed
+func setupCommonFlags(subcommand *flag.FlagSet) {
+	subcommand.StringVar(&backupNS, "namespace", defaultNamespace, "The namespace the backup and restore should operate.")
+	subcommand.BoolVar(&verify, "verify-succeeded", false, "Check that the operation completed successfully.")
+}
+
+//TODO: update flags default if needed
+func setupBackupFlags(backupCmd *flag.FlagSet) {
+	backupCmd.StringVar(&includedNS, "include-namespaces", "*", "Namespaces to include in the backup (use '*' for all namespaces).")
+	backupCmd.StringVar(&selector, "selector", "", "Only back up resources matching this label selector.")
+	backupCmd.StringVar(&includeResources, "include-resources", "", "Resources to include in the backup, formatted as resource.group, such as storageclasses.storage.k8s.io (use '*' for all resources).")
+	backupCmd.StringVar(&snapshotLocation, "volume-snapshot-Location", "", "List of locations (at most one per provider) where volume snapshots should be stored.")
```

Comment:
Is this param maybe a little too velero specific?  Is it something that can be configured at a global level?  I think it would be ideal if the script/program has as few params as possibl

Reply:
this parameter will be passed from the tests,currently the tests dont really configure it.. another provider can just ignore it.